src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -26,8 +26,59 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 {
-    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
-    InstructionSet isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    NamedIntrinsic      intrinsicID = node->gtHWIntrinsicId;
+    InstructionSet      isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    HWIntrinsicCategory category    = compiler->categoryOfHWIntrinsic(intrinsicID);
+
+    if (category == HW_Category_SimpleSIMD)
+    {
+
+        GenTree*  op1        = node->gtGetOp1();
```

Comment:
When we start adding containment support, doing the operand handling here is going to cause a lot of duplicated handling.

We probably want to have a `genHWIntrinsic_<encoding>` method (e.g. `genHWIntrinsic_R_RM`) so we can have the correct handling based on the supported encodings (it also lets us handle cases, like the non-AVX instructions, where they are actually `RM_RM`, due to the additional `movaps` instruction that may need to be emitted).

Reply:
I plan to deal with containment with additional mask bit of the categories.